kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/BaseContentService.java

```diff
@@ -7015,11 +7015,42 @@ protected void handleAccessResource(HttpServletRequest req, HttpServletResponse
 					return; 
 				}
 
+				// If there is a direct link to the asset, no sense streaming it.
+				// Send the asset directly to the load-balancer or to the client
+				URI directLinkUri = m_storage.getDirectLink(resource);
```

Comment:
So if (resource instanceof WrappedContentResource) the rest of the code below will break .....

@botimer or @buckett any hints why?